storage_service/locations/models/space.py

```diff
@@ -513,7 +513,7 @@ def create_rsync_directory(self, destination_path, user, host):
         for directory in directories:
             path = os.path.join(os.path.dirname(directory), '')
             path = "{}@{}:{}".format(user, host, utils.coerce_str(path))
-            cmd = ['rsync', '-vv', '--protect-args', '--chmod=ugo+rw', '--recursive', temp_dir, path]
+            cmd = ['rsync', '-vv', '--protect-args', '--chmod=ugo+rwx', '--recursive', temp_dir, path]
```

Comment:
I don't think you missed anything the function you suggest looks good.  Do you want to update your PR to include it?